pytorch_lightning/plugins/precision/deepspeed.py

```diff
@@ -35,6 +36,13 @@ class DeepSpeedPrecisionPlugin(PrecisionPlugin):
     """Precision plugin for DeepSpeed integration."""
 
     def __init__(self, precision: Union[str, int], amp_type: str, amp_level: Optional[str] = None) -> None:
+        supported_precision_values = (16, 32, PrecisionType.BFLOAT.value, PrecisionType.MIXED.value)
```

Comment:
why is that?

Reply:
it's because in PrecisionType `16` is marked as `"16"`. same for `32`, so I am not sure why it's done like that. I have seen a lot of places in the code-base where we are not using enums nor comparing the values with enums. Just using hard-coded `16`.

That's why I kept it here like this for reviewers to get an idea of how it will end up in the future.

Reply:
The `LightningEnum` compares with the string value anyways so we don't need to make this distinction here

https://github.com/PyTorchLightning/pytorch-lightning/blob/dbd69b9a09f3954297d78996389207501e96cd1b/pytorch_lightning/utilities/enums.py#L36-L38

Reply:
done!